### PR TITLE
Feat/edit page

### PR DESCRIPTION
**Pull request by yoslie**
### Overview
create edit page

** Todo: create function save edit, bind data store to widget

### Changes
- add page module getPageById
- create page edit halaman
- update page aside component last update and status

### Preview

https://github.com/jabardigitalservice/j-site-builder/assets/65396086/4a1975c5-a9c1-4471-8e0a-42aa9c47a8d9

### Evidence
title: create edit page and set initiate data
project: J-Site
participants: @yoslie @Ibwedagama @doohanas @rizfirman @azhargraha @marsellavaleria19

### CODE REVIEW

File: repository/j-site/modules/page.ts [hunk 1]
```diff
@@ -27,6 +28,22 @@ class PageModules extends FetchFactory {
...
+    idPage: string,
+    fetchOptions?: FetchOptions<'json'>,
+    options?: AsyncDataOptions<IPageDetailResponse>,
+  ) {
+    return useAsyncData(() => {
+      return this.call<IPageDetailResponse>(
+        'GET',
+        `${this.RESOURCE}/${idSetting}/${idPage}`,
+        undefined, // body
+        fetchOptions,
```
[the remark this answers is not in the document]
Comment (yoslie) [reply repository/j-site/modules/page.ts]: sure changed [here]([refactor: remove unused param](https://github.com/jabardigitalservice/j-site-builder/pull/144/commits/7f36f57b2fbb5d7f3b27cf96f47bb8f8771c9009))

File: pages/halaman/ubah/[id].vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  async function setInitialData() {
+    const pageId = route.params.id.toString()
+
+    const {
+      data: dataPage,
+      status,
+      error,
+      pending,
+    } = await $jSiteApi.page.getPageById(siteStore?.siteId ?? '', pageId)
```
[the remark this answers is not in the document]
Comment (yoslie) [reply pages/halaman/ubah/[id].vue]: thanks, changed here [refactor: add server false into fetch page by id](https://github.com/jabardigitalservice/j-site-builder/pull/144/commits/86b1b80f8b9e1d59bfbec8fbcb2a2116dd62b100)

File: repository/j-site/types/page.ts [hunk 3]
```diff
@@ -37,6 +37,11 @@ export interface IPagesResponse {
   meta?: IMetaData
 }
 
+export interface IPageDetailResponse {
+  data?: IPageDataExtended
+  meta?: IMetaData
```
[the remark this answers is not in the document]
Comment (yoslie) [reply repository/j-site/types/page.ts:42]: just following the others 🤣 is it unnecesary?

**Review by doohanas — APPROVED** [review]

File: repository/j-site/types/page.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ibwedagama) [reply repository/j-site/types/page.ts:42]: it's okay @yoslie it's just a lil bit funny 😆

**Review by Ibwedagama — APPROVED** [review]

**Review by rizfirman — APPROVED** [review]

**Review by azhargraha — APPROVED** [review]

**Review by marsellavaleria19 — APPROVED** [review]

**Pull request merged**